lib/expansion_rules.rb

```diff
@@ -148,21 +148,30 @@ def dependency_resolution
     @dependency_resolution ||= ExpansionRules::DependencyResolution.new(self)
   end
 
-  def find_custom_expansion_fields(document_type, options = {})
-    should_check_link_type = options[:link_type]
-    link_type = options[:link_type].try(:to_sym)
+  def find_custom_expansion_fields(document_type, link_type = nil)
+    matching_document_types = CUSTOM_EXPANSION_FIELDS.select do |item|
+      item[:document_type] == document_type.to_sym
+    end
 
-    condition = CUSTOM_EXPANSION_FIELDS.find do |cond|
-      next if should_check_link_type && cond.fetch(:link_type, link_type) != link_type
+    return nil unless matching_document_types.any?
 
-      cond[:document_type] == document_type.to_sym
-    end
+    if link_type.nil?
+      collated_fields = matching_document_types.inject([]) { |memo, item| memo + item[:fields] }
+      may_use_default_fields = matching_document_types.all? { |item| item[:link_type].present? }
 
-    condition[:fields] if condition
+      collated_fields += DEFAULT_FIELDS if may_use_default_fields
+      collated_fields.uniq
```

Comment:
Are we testing the `.uniq` behaviour anywhere in "expansion_rules_spec.rb"?

Reply:
Got that covered now, thanks